scrapy/utils/curl.py

```diff
@@ -21,14 +25,24 @@ def error(self, message):
 curl_parser.add_argument('-d', '--data', dest='data')
 curl_parser.add_argument('-u', '--user', dest='auth')
 
-curl_parser.add_argument('--compressed', action='store_true')
-curl_parser.add_argument('-s', '--silent', action='store_true')
 
+safe_to_ignore_arguments = [
+    ['--compressed'],
+    ['-s', '--silent'],
+    ['-v', '--verbose'],
+    ['-#', '--progress-bar']
+]
+
+for argument in safe_to_ignore_arguments:
+    curl_parser.add_argument(*argument, action='store_true')
```

Comment:
Not sure about this. IMO those arguments we know that can be ignored shouldn't raise an error even when passing `ignore_unknown_options=False` as we know that they don't affect to the Scrapy request and could be really common (i.e.: --compress, which used to be in the cURL commands coming from the network tool in browsers).

@kmike I can delete this or maintain it here. Up to you.

Reply:
I like an idea of having such list.

It seems however that `--compressed` is not safe to ignore, as it affects request headers.

Reply:
I based that on the original work and I also saw that this tool: https://curl.trillworks.com/ ignore it. 

However, I've seen that using `--compressed` cURL adds the `Accept-Encoding: deflate, gzip` header. 

Do you think I should implement that?

Reply:
Scrapy adds this header by default (see https://github.com/scrapy/scrapy/blob/master/scrapy/downloadermiddlewares/httpcompression.py), so I'm not sure what to do.

Without this option we shouldn't set this header, but it can be tricky to implement here - having an empty header is not the same, if I'm not mistaken.

So a strict way would be probably to show a warning (a different one) if `--compressed` is not passed? And then implement a way to disable setting Accept header per-request in HttpCompressionMiddleware, and use it here.

Reply:
we may have a similar issue with UserAgentMiddleware, DefaultHeadersMiddleware and other middlewares which set headers. Not sure how big is an issue though, maybe a warning in docs is fine.

Reply:
Ok, so I added a "caution" message in docs.
![captura_13](https://user-images.githubusercontent.com/22377678/62061009-9207d900-b226-11e9-9212-fcb75e5265f0.png)

Maybe that should be enough to start with (?).

Reply:
Could you please add a comment to `--compressed` item in the ignored list, which explains that it is not safe to ignore, but ignored because this is Scrapy default?

Reply:
Of course! I did it :+1: